lua/bufferline.lua

```diff
@@ -94,19 +99,19 @@ function bufferline.enable()
   create_autocmd(
     {'BufEnter', 'BufWinEnter', 'BufWinLeave', 'BufWipeout', 'BufWritePost', 'SessionLoadPost', 'VimResized', 'WinEnter', 'WinLeave'},
     {
-      callback = bufferline.update,
+      callback = function() bufferline.update() end,
       group = augroup_bufferline_update,
     }
   )
 
   create_autocmd('OptionSet', {
-    callback = bufferline.update,
+    callback = function() bufferline.update() end,
     group = augroup_bufferline_update,
     pattern = 'buflisted',
   })
 
   create_autocmd('WinClosed', {
-    callback = bufferline.update_async,
+    callback = function() bufferline.update_async() end,
```

Comment:
Why these 3 changes?

Reply:
When assigning something to be the `callback` of `nvim_create_autocmd`, that callback will receive a single argument which is a table having all data related to the specific event which was triggered. The functions I replaced with closures have actual arguments (e.g. `update_async` accapts `update_names`, a bool) and we could unintentionally introduce bugs to these functions should we ever expect function arguments matching the function signature, and receive the table from `nvim_create_autocmd` instead.